test.el

```diff
@@ -275,11 +275,34 @@ that buffer once BODY finishes."
   (with-temp-buffer
     (insert-file-contents
      (expand-file-name "testdata/fill.BUILD" bazel-test--directory))
-    (bazel-mode)
-    (search-forward "# The Foobar files")
-    (let ((before (buffer-string)))
-      (fill-paragraph)
-      (should (equal (buffer-string) before)))))
+    (let ((original-buffer (current-buffer)))
+      (dolist (comment-text
+               '("keep sorted"
+                 "KEEP SORTEd"
+                 "do not sort"
+                 "@unused"
+                 "@unsorted-dict-items"
+                 "buildifier: leave-alone"
+                 "buildifier: disable=foo"
+                 "buildifier: disable=confusing-name"
+                 "BUILDIFIER: DISABLE=CONFUSING-NAME"
+                 "buildozer: disable=git-repository"
+                 ))
+        (ert-info
+            ((format "Magic comment '%s' not kept on its own line" comment-text))
+          (with-temp-buffer
+            (insert-buffer-substring original-buffer)
+            (goto-char (point-min))
+            (bazel-mode)
+
+            (while (search-forward "%MAGIC_COMMENT%" nil t)
```

Comment:
It would be better to keep the magic comments literally in the test file.

We do perform similar replacements in other tests, but these are for directory names like the test workspace directory that are necessarily arbitrary. Here I think it's better to keep tests as logic-free as possible. I.e., I'd introduce multiple paragraphs into the test file with different magic comments and run `fill-paragraph` on all of them.

There are probably diminishing returns to try to test all combinations. I'd rather pick a few "important" examples and otherwise assume that regex matching works as expected. E.g. if `@unused` is matched literally, so will `@unsorted-dict-items`.

Reply:
Fair enough.  Tests updated accordingly.

One thing to think about is that while filling keeps paragraphs like this intact:

```
    # Test paragraph
    # keep sorted
```

But paragraphs like these:

```
    # keep sorted
    # Test paragraph
```

Are filled to:
```
    # keep sorted Test paragraph
```

And

```
    # The magic comment here should be kept on the end of its own line but keep sorted isn’t
```

is filled to 
```
    # The magic comment here should be kept on the end of its own line but keep
    # sorted isn’t
```

Empirically, `buildifier` doesn't seem to care where magic comments are on a line, so they probably shouldn't be split like this.  It looks like [`fill-nobreak-predicate`](https://www.emacswiki.org/emacs/FillParagraph#h5o-3) could be used for this purpose.

Thoughts? I can look into this, if you'd like.  I think I'll have some time next week to try it out.